### PR TITLE
Refactor profile update action

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
## Notion card
https://www.notion.so/6afc9500b2cc42af86be55f37894fb09?v=185bd45b9dc24dfcad9c57c07d32a7c0&p=51ad6bf864c448308db7a9c6f65241ad&pm=s

## Summary

- Created `UpdateProfileSettingsService` service and moved all the logic related to `profile_controller#update` action to it.
- Fixed the Axios error toast message. Earlier on the frontend, Axios is not handling error messages but instead displays the request status when the error message from the backend is not an array of messages.
- Now on the profile settings error message are displayed properly with the above-mentioned change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by apoorv1316** [comment]
@swaathi Please review it.

File: app/services/update_profile_settings_service.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class UpdateProfileSettingsService
+  attr_reader :current_user, :user_params
+
+  def initialize(current_user, user_params)
+    @current_user = current_user
+    @user_params = user_params
+  end
+
+  def process
```
[the remark this answers is not in the document]
Comment (prasanthchaduvula) [reply app/services/update_profile_settings_service.rb:11]: Removed all the custom validations which were added as the devise is already handling them.

File: app/services/update_profile_settings_service.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  def validate_password_confirmation
+    return true if user_params[:password] == user_params[:password_confirmation]
+
+    raise Exception.new("Password and password confirmation does not match")
+  end
+
+  def validate_password_length
+    return true if user_params[:password].present? &&
+    user_params[:password].length > 5 &&
```
[the remark this answers is not in the document]
Comment (prasanthchaduvula) [reply app/services/update_profile_settings_service.rb]: Devise handles the password validations along with the length of the password. Removed all the custom validations which were added as the devise is already handling them.

**Comment by prasanthchaduvula** [comment]
@swaathi Please review

**Pull request merged**